### PR TITLE
Workflow, which builds tools on Mac OS and Windows

### PR DESCRIPTION
**Pull request by gzigzigzeo**

### CODE REVIEW

File: .github/workflows/build-macos.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: macos-12
+
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout Teleport
+        uses: actions/checkout@v3
+
+      - name: Setup Go
```
Comment (zmb3) [inline .github/workflows/build-macos.yaml:20]: Should we install rust too? If we don't, desktop access won't be built

File: .github/workflows/build-windows.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Build on Windows
+run-name: Build on Windows
+
+on:
+  pull_request:
+  push:
+  
+jobs:
+  build:
+    name: Build all tools on Windows
```
Comment (zmb3) [inline .github/workflows/build-windows.yaml:10]:
```suggestion
    name: Build tsh on Windows
```

File: .github/workflows/build-macos.yaml (hunk 1 above)
Comment (gzigzigzeo) [reply .github/workflows/build-macos.yaml:20]: MacOS 12 and windows-latest images have the latest Rust installed by default: https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md Default go version is 1.17, though. That's the reason why we set it up separately.

File: .github/workflows/build-macos.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+run-name: Build on Mac OS
+
+on:
+  pull_request:
+  push:
+  
+jobs:
+  build:
+    name: Build all tools on Mac OS
+    runs-on: macos-12 # macos-lates is Big Sur for now
```
Comment (zmb3) [inline .github/workflows/build-macos.yaml:11]:
```suggestion
    runs-on: macos-12 # macos-latest is Big Sur for now
```

**Review by zmb3 — APPROVED** [review]
Looks good, though it would be nice if we could find a way to not have to keep `go-version` in sync with the rest of the project.

File: .github/workflows/build-macos.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+name: Build on Mac OS
+run-name: Build on Mac OS
+
+on:
+  pull_request:
```
Comment (r0mant) [inline .github/workflows/build-macos.yaml:5]: I've just realized that we probably should only be invoking these workflows if there are code changes, similar to how we only run unit and other tests based on path filters? I can make this changes as a part of the [buddy PR](https://github.com/gravitational/teleport/pull/19226).

**Comment by r0mant** [comment]
Merged in https://github.com/gravitational/teleport/pull/19226.

**Pull request closed without merging**